FormHandler/AbstractMessageFormHandler.php

```diff
@@ -22,9 +23,9 @@
     protected $sender;
     protected $participantProvider;
 
-    public function __construct(Request $request, ComposerInterface $composer, SenderInterface $sender, ParticipantProviderInterface $participantProvider)
+    public function __construct(RequestStack $requestStack, ComposerInterface $composer, SenderInterface $sender, ParticipantProviderInterface $participantProvider)
     {
-        $this->request = $request;
+        $this->request = $requestStack->getCurrentRequest();
```

Comment:
The request should not be retrieved in the constructor.
You can do something like:
```php
/**
 * @param Request|RequestStack $request
 */
public function __construct($request, ...) {
    if ($request instanceof Request) {
         @trigger_error(sprintf('Using an instance of "%s" as first parameter of "%s" is deprecated since version 1.3 and won\'t be supported in 2.0. Use an instance of "Symfony\Component\HttpFoundation\RequestStack" instead.', get_class($request), __METHOD), E_USER_DEPRECATED);
    }
    $this->request = $request;
    // ...
}

private function getCurrentRequest() {
   if ($this->request instanceof Request) {
      return $this->request;
   }
   
   return $this->request->getCurrentRequest();
}

public function process() {
    // ...
    $request = $this->getCurrentRequest();
}